src/components/Avatar/Avatar.test.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import { Avatar, AvatarProps } from './';
+
+const DEFAULT_PROPS: AvatarProps = {
+  size: 'regular',
+  type: 'circle'
+};
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+describe('when using default props', () => {
+  test('should be regular size', () => {
+    const { container } = render(<Avatar {...DEFAULT_PROPS} />);
+    expect(container.firstChild).toHaveAttribute('data-size', 'regular');
+  });
+
+  test('should be medium size', () => {
+    const { container } = render(<Avatar {...DEFAULT_PROPS} size="medium" />);
+    expect(container.firstChild).toHaveAttribute('data-size', 'medium');
+  });
+
+  test('should be circle shape', () => {
+    const { container } = render(<Avatar {...DEFAULT_PROPS} type="circle" />);
+    expect(container.firstChild).toHaveAttribute('data-type', 'circle');
+  });
+
+  test('should be square shape', () => {
+    const { container } = render(<Avatar {...DEFAULT_PROPS} type="square" />);
+    expect(container.firstChild).toHaveAttribute('data-type', 'square');
+  });
+});
+
+describe('when status type is "active"', () => {
+  test('should show status active', () => {
+    const { container } = render(<Avatar {...DEFAULT_PROPS} statusType="active" />);
+    expect(container.querySelector('[data-status-type="active"]')).toBeInTheDocument();
+  });
+});
+
+describe('when badge content is "9+"', () => {
+  test('should show badge', () => {
+    const { getByText } = render(<Avatar {...DEFAULT_PROPS} badgeContent="9+" />);
+    expect(getByText('9+')).toBeInTheDocument();
+  });
+});
```

Comment:
Oh, I missed a test case in my last review. Can we also write a test to ensure a user's initials are visible if there's no badge content. Can we also write tests to make sure different name lengths work, i.e. `John` appears as `J`, `John Smith` appears as `JS`, and `John Smith Appleseed` appears as `JA`?

Reply:
Added the missed test cases. We are always displaying two letters from the user name. As per figma if there are three words we are fetching the first letters from the first two words.
<img width="185" alt="Screenshot 2023-02-13 at 21 39 16" src="https://user-images.githubusercontent.com/27946024/218633198-05a940f0-2200-4f97-b445-fd778509ff47.png">

Reply:
Awesome, good stuff!